src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostConfiguration.cs

```diff
@@ -0,0 +1,187 @@
+
+        /// <summary>
+        /// Returns the sources used to obtain configuration values.
+        /// </summary>
+        public IList<IConfigurationSource> Sources => _sources.ToList();
+
+        /// <summary>
+        /// Returns the providers user to obtain configuration values.
+        /// </summary>
+        public IEnumerable<IConfigurationProvider> Providers => _providers.ToList();
+
+        /// <summary>
+        /// Gets a key/value collection that can be used to share data between the <see cref="IConfigurationBuilder"/>
+        /// and the registered <see cref="IConfigurationProvider"/>s.
+        ///
+        /// In this implementation, this largely exists as a way to satisfy the
+        /// requirements of the IConfigurationBuilder and is not populated by
+        /// <see cref="WebAssemblyHostConfiguration"/> with any meaningful info.
+        /// </summary>
+        IDictionary<string, object> IConfigurationBuilder.Properties { get; } = new Dictionary<string, object>();
+
+        /// <inheritdoc />
+        public string this[string key]
+        {
+            get {
+                // Iterate through the providers in reverse to extract
+                // the value from the most recently inserted provider.
+                for (var i = _providers.Count - 1; i >= 0; i--)
+                {
+                    var provider = _providers[i];
+
+                    if (provider.TryGet(key, out var value))
+                    {
+                        return value;
+                    }
+                }
+
+                return null;
+            }
+            set {
+                if (_providers.Count == 0)
+                {
+                    throw new InvalidOperationException("Can only set property if at least one provider has been inserted.");
+                }
+
+                foreach (var provider in _providers)
+                {
+                    provider.Set(key, value);
+                }
+
+            }
+        }
+
+        /// <summary>
+        /// Gets a configuration sub-section with the specified key.
+        /// </summary>
+        /// <param name="key">The key of the configuration section.</param>
+        /// <returns>The <see cref="IConfigurationSection"/>.</returns>
+        /// <remarks>
+        ///     This method will never return <c>null</c>. If no matching sub-section is found with the specified key,
+        ///     an empty <see cref="IConfigurationSection"/> will be returned.
+        /// </remarks>
+        public IConfigurationSection GetSection(string key) => new ConfigurationSection(this, key);
+
+        /// <summary>
+        /// Gets the immediate descendant configuration sub-sections.
+        /// </summary>
+        /// <returns>The configuration sub-sections.</returns>
+        public IEnumerable<IConfigurationSection> GetChildren()
+        {
+            return Providers
+                .Aggregate(Enumerable.Empty<string>(),
```

Comment:
You might be able to use `SelectMany` here instead of `Aggregate`

`Providers.SelectMany(s => source.GetChildKeys(Enumerable.Empty<string>(), null))`